Map/Source/SquareMap.cpp

```diff
@@ -62,4 +63,133 @@ SquareMap::SquareMap(const int mapSize)
     {
         _fieldList.at(i).at(0) = std::make_unique<FightField>();
     }
-}
\ No newline at end of file
+}
+
+SquareMap::SquareMap(FieldList&& fieldList) : _fieldList(std::move(fieldList))
+{
+}
+
+bool SquareMap::isFieldAccessible(const Position& position)
+{
+    if (not isField(position) or getField(position)->getType() == FieldType::Wall)
+    {
+        return false;
+    }
+    return true;
+}
+
+bool SquareMap::isField(const Position& position)
```

Comment:
jak tobisz try na cala metode, to ladniejszym rozwiazaniem jest  bool SquareMap::isField(const Position& position) try
{
}
catch
{
}